### PR TITLE
refactor filter logic in DexScreen

### LINKED ISSUE #2340: Thanks for following up, added a few comments.
Thanks for following up, added a few comments. 
As for refactoring the repeated logic, we are more than happy if you wish to contribute in the same PR!
Otherwise, you could file an issue and we would address it in the future

_Originally posted by @kyleleow in https://github.com/DeFiCh/wallet/pull/2300#pullrequestreview-901214403_

### PR DESCRIPTION
**Pull request by rox1338**
#### What kind of PR is this?:
/kind refactor

#### What this PR does / why we need it:
Refactor the filter logic in DexScreen.tsx and PoolPairCards.tsx
It should only be implemented once and reused.

#### Which issue(s) does this PR fixes?:
Fixes #2340 

#### Developer Checklist:
- [x ] Read your code changes at least once
- [x ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

**Comment by rox1338** [comment]
@kyleleow 
We discussed this in #2300 
Could you have a look?

**Comment by kyleleow** [comment]
Hi @rox1338 thanks for creating a PR, after testing on my device, the sorting for pool pairs under `Your pool pairs` doesn't follow the ascending order of `id` like how it is used to.

I haven't gone deep dive into the code yet what went wrong, perhaps you can check it again?

File: mobile-app/app/screens/AppNavigator/screens/Dex/DexScreen.tsx [hunk 1]
```diff
@@ -112,34 +112,77 @@ export function DexScreen (): JSX.Element {
     })
   }
 
+  // sorting functions
+  const availablePairsSortingFn = (firstPair: DexItem<PoolPairData>, secondPair: DexItem<PoolPairData>): number => (
+    new BigNumber(secondPair.data.totalLiquidity.usd ?? 0).minus(firstPair.data.totalLiquidity.usd ?? 0).toNumber() ??
+    new BigNumber(secondPair.data.id).minus(firstPair.data.id).toNumber()
+  )
+  const yourPairsSortingFn = (firstPair: DexItem<WalletToken>, secondPair: DexItem<WalletToken>): number => (
+    pairs.findIndex(x => x.data.id === firstPair.data.id) -
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Dex/DexScreen.tsx:121]: I have just clarified with @pierregee that was previously working on this logic, the sorting for `Your pool pairs` was meant to follow the order of `id` in the pairs sorted in `Available pool pairs`.

```suggestion
    filteredAvailablePairs.findIndex(x => x.data.id === firstPair.data.id) -
```

**Review by kyleleow — COMMENTED** [review]
Thanks @rox1338 i will review again when i have more time

File: mobile-app/app/screens/AppNavigator/screens/Dex/DexScreen.tsx (hunk 1 above)
Comment (rox1338) [reply mobile-app/app/screens/AppNavigator/screens/Dex/DexScreen.tsx]: Thanks for the clarification.
I implemented a fix.
I introduced an intermediate variable called sortedAvailablePairs which is never filtered to base the sorting on.

**Pull request merged**